protocol/communities/manager.go

```diff
@@ -4771,6 +4956,70 @@ func (m *Manager) createCommunityTokenPermission(request *requests.CreateCommuni
 	}
 
 	return community, changes, nil
+
+}
+
+func (m *Manager) UpdateControlNode(communityID types.HexBytes, controlNode types.HexBytes) (*Community, error) {
+	community, err := m.GetByID(communityID)
+	if err != nil {
+		return nil, err
+	}
+	if community == nil {
+		return nil, ErrOrgNotFound
+	}
+
+	pubKey, err := crypto.UnmarshalPubkey(controlNode)
+	if err != nil {
+		return nil, err
+	}
+
+	if community.ControlNode().Equal(pubKey) {
+		return community, nil
+	}
+
+	community.setControlNode(pubKey)
+	err = m.persistence.SaveCommunity(community)
+	if err != nil {
+		return nil, err
+	}
+
+	return community, nil
+}
+
+func (m *Manager) UpdatePrivateKeyAndControlNode(communityID types.HexBytes, pk *ecdsa.PrivateKey) (*Community, error) {
+	community, err := m.UpdatePrivateKey(communityID, pk)
+	if err != nil {
+		return nil, err
+	}
+	keyBytes := common.PubkeyToHexBytes(&pk.PublicKey)
```

Comment:
Maybe I do not see something but does it make sense to marshal pubkey here and then unmarshal in UpdateControlNode?